pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go

```diff
@@ -70,6 +70,10 @@ type streamIngestionFrontier struct {
 	// persistedReplicatedTime stores the highwater mark of
 	// progress that is persisted in the job record.
 	persistedReplicatedTime hlc.Timestamp
+	// heartbeatTime is the minimum timestamp of persistedReplicatedTime
```

Comment:
nit: this comment could instead be: `heartbeatTime represents the earliest timestmp for which...`. I don't think we need to explain what it is set to, as the reader can infer this from the code.